### DISCUSSION TITLE
MQTT topic for runtime/ cost/ kWh

### DISCUSSION THREAD

**Discussion by RGarrett93** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#128] [Q&A] [answered]
Hi,
Is it possible to get the runtime/ cost via MQTT? 
I tried several different Topics but couldn't find any with "CONTENT":"TIMES".
I want to create an entity using the MQTT data, then work out total kWh consumption - I can then input that in to the new Home Assistant 'Energy' monitoring.
Thanks

**Comment by visualapproach** [comment, marked as answer]
Sure you can! Go to HA configuration, integrations, mqtt config and listen to topic `BW_2.0.0/times` and you get this:
![image](https://user-images.githubusercontent.com/22987250/136572530-45c46461-fb0d-4f2a-bfe2-4c12be477167.png)

**Reply by RGarrett93** [reply]
That's great, thanks for that: glad it is possible.

I've tried subscribing to 'BW_2.0.0/times' but nothing is appearing for that topic after a duration of waiting.
How often is that topic published?

I've tested 'BW_2.0.0/message' and get those messages through fine.
I believe I've got the latest development branch installed via OTA.
Could that be why, should I stick with the latest Master branch v2.1.3 instead?

Thanks
Ryan

**Reply by visualapproach** [reply]
Dev branch is fine. Publishes every 10 minutes plus when something changes. Try forcing it by starting pump or similar.

**Reply by RGarrett93** [reply]
Yeah I've tried switching a few things but nothing coming through yet.
I get messages everytime for BW_2.0.0/message, so I know my MQTT is working.

Does the master branch have the BW_2.0.0/times incorporated, I could try uploading that one instead?

**Reply by visualapproach** [reply]
Think so but I know it is working in dev

**Reply by RGarrett93** [reply]
It's working now, I uploaded it a few times but to no avail then I tried 'Verbose upload' and thereafter BW_2.0.0/times started coming through.

Thanks for your help 👍

**Comment by RGarrett93** [comment]
### Home Assistant energy usage

**Add to 'configuration.yaml'**
```
  - platform: mqtt
    name: "Lazy Spa Total Consumption"
    state_topic: "BW_2.0.0/times"
    value_template: "{{ value_json.COST }}"    
    unit_of_measurement: 'kWh'
    availability_topic: "BW_2.0.0/Status"
    payload_available: "Alive"
    payload_not_available: "Dead"
    state_class: total_increasing
    device_class: energy
    icon: 'mdi:flash'
```

I've temporary changed the config cost value to 0.01 so that I can quickly work out the total kW usage, eventually I will get HA to calculate it all: or maybe we could have a seperate "KWH" under "COST" in "CONTENT": "TIMES".

You can then add your lazy spa as an individual device in the energy tab which will break down the usage with the rest of your appliances.



![overall usage](https://user-images.githubusercontent.com/60946160/136702794-25f6abfb-b5fc-4b97-afa1-52ef7008e19a.png)
![Usage hot tub](https://user-images.githubusercontent.com/60946160/136702801-bcfbc225-80d0-4058-acf7-f25582c7dea0.png)

**Comment by visualapproach** [comment]
In development branch, "KWH" is sent in the "times" topic.
